File: Leetcode/1480. Running Sum of 1d Array/solution.rs
impl Solution {
    pub fn running_sum(nums: Vec<i32>) -> Vec<i32> {
        let mut counter = nums[0];
        let mut answer = vec![counter];
        
        for i in 1..nums.len() {
            counter += nums[i];
            answer.push(counter);
        }
        
        answer
    }
}

File: Leetcode/1480. Running Sum of 1d Array/solution.cpp
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> answer;
        answer.push_back(nums[0]);
        
        for(int i = 1; i < nums.size(); i++) {
            answer.push_back(nums[i] + answer[answer.size() - 1]);
        }
        
        return answer;
    }
};